interview-questions.md

```diff
@@ -8,50 +8,59 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Your answer:
+Your answer: Equality, Logical, Relational operators will all return an evaluation of a boolean data type.
+
+Researched answer: the ! or bang boolean operator is evaluating if condition is the opposite of what is being compared.  It is not returning the direct true or false boolean logical value of the condition being evaluated.
 
-Researched answer:
 
 3. What is an index? What is the difference between index and value?
 
-Your answer:
+Your answer: Index is the position of character of a String data type. When looking at Arrays it is the postion of the element within the array.  Counting index position's within Strings and Arrays always starts with position 0.  Value of a string is defined as the characters of a string, and in an array the values are the indivudal elements data types.
 
-Researched answer:
+Researched answer: Strings are ordered sequences of character data. Indexing allows you to access individual characters in a string. A string value is just a sequence of characters such as letters, numbers. A string value uses quotes syntax. 
+An array is a structure that contains an ordered collection of data types, within this context they are called elements.  each element can be referenced by its index value in the array. 
 
 4. Compare and contrast arrays and strings. Please include ways in which they are the same and ways in which they are different.
 
-Your answer:
+Your answer: Similarities: You can refrence the length of a string and array.  When accessing the of characters, an index value is used to log the position.  Both can be used to describe an content of its subject. Strings are used to discribe the subject, while arrays can contain its subject as diffrent forms of data types.
+Diffrences: strings 
 
-Researched answer:
+Researched answer: Arrays siz are fixed base on the content when assigned. When assigining a string the character size can vary.  Both strings and arrays can be manipulated or accessed through its properties and built in method.
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-Your answer:
+Your answer: Pair programing involves contributions of two developers.  While one is manipulating the code and making changes based of calls the navigator is making. They control and maintain the active workspace. This developer is in the role as Driver.  The Navigator's contribution is the overviewer.  They contribute through adding input of ideads, brainstorm, syntax and grammer correctioss.  The nagivator is not activly engaged with the keyboard, because of this reserve role they are afforded more room to focus on the big picture.  It is important for both developers to perform in both of these roles to understand the perspective of hands on, actively engaged, and reserve overview.  Big picture, Little picture.  This wholeistic approach better insures complete best practice execution is 
+attaianed when developing.
```

Comment:
Awesome initial answer! Love that you mention both developers performing both roles. 👍🏽